src/Microsoft.AspNetCore.Hosting/WebHostExtensions.cs

```diff
@@ -90,8 +90,13 @@ public static void Run(this IWebHost host)
                     done.Wait();
                 };
 
+#if NETCOREAPP2_0
                 var assemblyLoadContext = AssemblyLoadContext.GetLoadContext(typeof(WebHostExtensions).GetTypeInfo().Assembly);
                 assemblyLoadContext.Unloading += context => shutdown();
```

Comment:
Have we tried the `AppDomain.CurrentDomain.ProcessExit` event? This is available in .NET Standard 2.0. On .NET Core 2.0, this event is analogous to AppContext.ProcessExit.